internal/commands/root/root.go

```diff
@@ -166,37 +165,34 @@ func runRootCommandWithProviderAndClient(ctx context.Context, pInit provider.Ini
 		"watchedNamespace": c.KubeNamespace,
 	}))
 
-	var leaseClient v1beta1.LeaseInterface
-	if c.EnableNodeLease {
-		leaseClient = client.CoordinationV1beta1().Leases(corev1.NamespaceNodeLease)
-	}
-
 	nodeProvider, ok := p.(node.NodeProvider)
 	if !ok {
 		nodeProvider = node.NaiveNodeProvider{}
 	}
 	pNode := NodeFromProvider(ctx, c.NodeName, taint, p, c.Version)
-	nodeRunner, err := node.NewNodeController(
-		nodeProvider,
-		pNode,
-		client.CoreV1().Nodes(),
-		node.WithNodeEnableLeaseV1Beta1(leaseClient, nil),
-		node.WithNodeStatusUpdateErrorHandler(func(ctx context.Context, err error) error {
-			if !k8serrors.IsNotFound(err) {
-				return err
-			}
-
-			log.G(ctx).Debug("node not found")
-			newNode := pNode.DeepCopy()
-			newNode.ResourceVersion = ""
-			_, err = client.CoreV1().Nodes().Create(ctx, newNode, metav1.CreateOptions{})
-			if err != nil {
-				return err
-			}
-			log.G(ctx).Debug("created new node")
-			return nil
-		}),
-	)
+
+	var opts []node.NodeControllerOpt
+	if c.EnableNodeLease {
```

Comment:
Let's default this to true, since we're at it.

Reply:
done

Reply:
Thanks